sdk/state/payment_test.go

```diff
@@ -553,35 +602,30 @@ func TestLastConfirmedPayment(t *testing.T) {
 
 	ca, err := sendingChannel.ProposePayment(200)
 	require.NoError(t, err)
+	assert.Equal(t, ca, sendingChannel.latestUnauthorizedCloseAgreement)
 
-	ca, authorized, err := receiverChannel.ConfirmPayment(ca)
-	assert.False(t, authorized)
+	caResponse, err := receiverChannel.ConfirmPayment(ca)
 	require.NoError(t, err)
-	assert.Equal(t, ca, receiverChannel.latestUnauthorizedCloseAgreement)
+	assert.Equal(t, caResponse, receiverChannel.latestAuthorizedCloseAgreement)
 
 	// Confirming a close agreement with same sequence number but different Amount should error
 	caDifferent := CloseAgreement{
 		Details: CloseAgreementDetails{
 			IterationNumber: 1,
 			Balance:         400,
 		},
-		CloseSignatures: ca.CloseSignatures,
+		DeclarationSignatures: ca.DeclarationSignatures,
+		CloseSignatures:       ca.CloseSignatures,
 	}
-	_, authorized, err = receiverChannel.ConfirmPayment(caDifferent)
-	assert.False(t, authorized)
+	_, err = sendingChannel.ConfirmPayment(caDifferent)
 	require.EqualError(t, err, "validating payment: close agreement does not match the close agreement already in progress")
-	assert.Equal(t, CloseAgreement{Details: CloseAgreementDetails{Balance: 0}}, receiverChannel.LatestCloseAgreement())
+	assert.Equal(t, ca, sendingChannel.latestUnauthorizedCloseAgreement)
 
 	// Confirming a payment with same sequence number and same amount should pass
-	ca, authorized, err = sendingChannel.ConfirmPayment(ca)
-	assert.True(t, authorized)
+	caFinal, err := sendingChannel.ConfirmPayment(caResponse)
 	require.NoError(t, err)
 	assert.Equal(t, CloseAgreement{}, sendingChannel.latestUnauthorizedCloseAgreement)
-
-	ca, authorized, err = receiverChannel.ConfirmPayment(ca)
-	assert.True(t, authorized)
-	require.NoError(t, err)
-	assert.Equal(t, CloseAgreement{}, receiverChannel.latestUnauthorizedCloseAgreement)
+	assert.Equal(t, caFinal, sendingChannel.latestAuthorizedCloseAgreement)
```

Comment:
💡 can we add:
```suggestion
	assert.Equal(t, caFinal, sendingChannel.latestAuthorizedCloseAgreement)
	assert.Equal(t, caFinal, caResponse)
```
testing that the final `Confirm` call is not changing the agreement but just storing the new sigs

Reply:
Good call.